src/replay_controller.hpp

```diff
@@ -79,7 +79,7 @@ class replay_controller : public events::observer
 
 	bool replay_ui_has_all_buttons() {
 		return play_button() && stop_button() && reset_button() &&
-		       play_turn_button() && play_side_button();
+		       play_turn_button() && play_side_button() && play_move_button();
```

Comment:
Does reset_replay_ui() need to be changed to call `play_move_button()->enable(true)`?

Reply:
Probably.

I'm now wondering about deleting replay_ui_has_all_buttons, and replacing it with a helper method that checks whether the button is null, and lets all the buttons that are available in the theme work.

Reply:
Hmm...
* replay_ui_playback_should_start is never called
* reset_replay_ui is never called
* update_replay_ui is a no-op, unless the theme is missing buttons

Reply:
Try looking in git history to see when they were last called (`git log -S replay_ui_playback_should_start src`, etc).

Reply:
Thanks, I'd been using gitk for that because I hadn't known of the -S option. Its callers were removed by the major rewrite 9d9422a4, "Instead of implementing its own game loop, replay_controller now uses play_controller game loop."

It seems all of the button::enable() calls are unnecessary logic with a necessary side-effect, because the hotkey system will enable or disable these buttons based on querying replay_controller::can_execute_command(). However, the enable() calls have the side-effect of triggering the hotkey system to check the buttons' required states and redrawing; commenting them out means that the buttons only update when another event triggers a redraw (such as hiding and showing the whole Wesnoth window).

How should the requery and redraw be triggered?

Reply:
There's a lot of seemingly unnecessary code, but I haven't figured out how trigger the requery properly. There's an attempt in https://github.com/stevecotton/wesnoth/tree/replay_hotkey_cleanup which has the replay controls working well, but the other UI buttons don't respond when the replay is paused.

Reply:
It's grown from a one-liner to 33 insertions(+), 155 deletions(-).